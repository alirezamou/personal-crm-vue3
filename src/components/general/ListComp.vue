<template>
  <ul v-if="any">
    <Item v-for="item in items" :key="item.id" :class="type">
      {{ item[itemKey] }}
    </Item>
  </ul>
</template>

<script>
import { h } from "vue";

export default {
  name: "ListComp",
  components: {
    Item: {
      name: "Item",
      render() {
        return h("li", this.$slots.default());
      },
    },
  },
  props: {
    items: {
      default: () => [],
      type: Array,
    },
    itemKey: {
      default: () => "value",
      type: String,
    },
    type: {
      default: () => "item",
      type: String,
    },
  },
  computed: {
    any() {
      return this.items.length > 0;
    },
  },
};
</script>
