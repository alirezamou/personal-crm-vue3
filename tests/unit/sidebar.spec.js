import { mount } from "@vue/test-utils";
import { createStore } from "vuex";
import Model from "@codeship/modelist";

import SidebarVue from "@/components/Sidebar.vue";
import ListComp from "@/components/general/ListComp.vue";

function factory(options = {}) {
  const stucture = {
    state: {},
    getters: {
      contacts: jest
        .fn()
        .mockReturnValue(new Model({ data: [{ name: "john" }] })),
    },
  };
  const store = createStore(stucture);
  return mount(SidebarVue, {
    global: {
      plugins: [store],
    },
    ...options,
  });
}

test("Sidebar renders the ListComp", () => {
  const wrapper = factory();
  expect(wrapper.findComponent(ListComp).exists()).toBe(true);
});
