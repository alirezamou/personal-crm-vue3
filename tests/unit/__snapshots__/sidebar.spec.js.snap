// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div id="sidebar" class="flex-1 min-w-1/4">
  <div class="flex justify-between text-sm mb-2 font-semibold"><span>Contacts: </span>
    <router-link to="/" class="!text-[#000] hover:no-underline hover:text-[#000]">+ new contact</router-link>
  </div>
  <ul>
    <li>
      <router-link to="[object Object]" class="contact">john</router-link>
    </li>
    <li>
      <router-link to="[object Object]" class="contact">jane</router-link>
    </li>
  </ul>
</div>
`;
